### PR TITLE
Translations/fetching [PIL-839]

### PR DESCRIPTION
**Pull request by vyvil**
PR for translations resources management.
Features' description can be found in readme file (attached to the PR).

There's still some TODOs left:
Caching management and versioning - now newest translation version from server is being fetched on app open and on language change and cached url is stored to cacheMap without any expiration date.

UI solution to change language. Now it takes user device's language and uses it if it's supported.
If not - uses default language.

N.B.:
In order for this feature to be testable during review period, I've set demo url that stores `en` and `lt` translations
`lt` version has only some strings translated - two on auth flow (for two buttons visible in the first screen) and four on common (navigation tab labels).
And I've temporarily added `lt` as supported language in order for those translations to work.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -281,8 +288,9 @@ class App extends React.Component<Props, *> {
                 }}
               />}
               {!!getEnv().SHOW_LANG_TOGGLE && <Button
-                title={`Change lang (current: ${i18n.language})`} // eslint-disable-line i18next/no-literal-string
-                onPress={() => changeLanguage(i18n.language === DEV_LANG ? DEFAULT_LANGUAGE : DEV_LANG)}
+                title={`Change lang (current: ${i18next.language})`} // eslint-disable-line i18next/no-literal-string
+                // eslint-disable-next-line i18next/no-literal-string
+                onPress={() => changeLanguage(i18next.language === 'lt' ? localeConfig.defaultLanguage : 'lt')}
```
Comment (poocart) [inline App.js:293]: Dev language will be `lt` now?  🇱🇹 👀

File: App.js [hunk 2]
```diff
@@ -328,16 +338,23 @@ const mapDispatchToProps = (dispatch: Dispatch): $Shape<Props> => ({
...
 
 const AppWithNavigationState = withTranslation()(connect(mapStateToProps, mapDispatchToProps)(App));
 
 const AppRoot = () => (
-  <Provider store={store}>
-    <PersistGate loading={<Container defaultTheme={defaultTheme}><LoadingSpinner /></Container>} persistor={persistor}>
-      {getEnv().SHOW_ONLY_STORYBOOK ? <Storybook /> : <AppWithNavigationState />}
-    </PersistGate>
-  </Provider>
+  <Suspense fallback={<View />}>
```
Comment (poocart) [inline App.js:348]: What if we add Spinner as fallback view as well, wdyt?

File: src/actions/authActions.js [hunk 3]
```diff
@@ -444,6 +445,10 @@ export const logoutAction = () => {
     await resetKeychainDataObject();
     if (themeType === DARK_THEME) await dispatch(setAppThemeAction(DARK_THEME)); // to persist dark theme after storage
     // is cleaned up so we would not blind users after they delete wallet :)
+
+    // leave translation initialisated flag in place
+    const areTranslationsInitialised = get(getState(), 'session.data.areTranslationsInitialised', false);
+    if (areTranslationsInitialised) dispatch(setSessionTranslationBundleInitialisedAction());
```
Comment (poocart) [inline src/actions/authActions.js:451]:
```suggestion
    if (getState()?.session?.data?.areTranslationsInitialised) {
      dispatch(setSessionTranslationBundleInitialisedAction());
    }
```

File: src/configs/localeConfig.js [hunk 4]
```diff
@@ -17,28 +17,32 @@
...
+
+const EN_COMMON = require('../locales/en/common.json');
+const EN_AUTH = require('../locales/en/auth.json');
+
+const DEFAULT_LANGUAGE = 'en';
+const DEFAULT_NAMESPACE = 'common';
+
+export default {
+  isEnabled: true,
+  baseUrl: 'http://pillar-stories.dev.imas.lt/locales/', // todo: change into real one;
```
Comment (poocart) [inline src/configs/localeConfig.js:36]: No real one yet?

File: App.js (hunk 1 above)
Comment (vyvil) [reply App.js:293]: Just for now. As mentioned in description ("N.B" part), I've added lt as supported language for testing. Will revert those changes after approval.

File: src/configs/localeConfig.js (hunk 4 above)
Comment (vyvil) [reply src/configs/localeConfig.js:36]: Nope - not yet.
When approved, I'll remove it so only locally stored translations will be used.
Kept it here for testing / reviews only :)

I'll update this part in separate PR as we've later on decided on versioning.
Expected solution is explained here: https://linear.app/pillarproject/issue/PIL-842/implementation-language-injection

**Comment by vyvil** [comment]
Closing it in regards of #2250

**Pull request closed without merging**